src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -65,17 +66,14 @@ pub(crate) fn collect_trait_impls(mut krate: Crate, cx: &mut DocContext<'_>) ->
         for &impl_def_id in tcx.trait_impls_in_crate(LOCAL_CRATE) {
             let mut parent = Some(tcx.parent(impl_def_id));
             while let Some(did) = parent {
-                attr_buf.extend(
-                    tcx.get_attrs(did, sym::doc)
-                        .filter(|attr| {
-                            if let Some([attr]) = attr.meta_item_list().as_deref() {
-                                attr.has_name(sym::cfg)
-                            } else {
-                                false
-                            }
-                        })
-                        .cloned(),
-                );
+                attr_buf.extend(tcx.get_all_attrs(did).iter().filter_map(|attr| match attr {
+                    Attribute::Parsed(AttributeKind::Doc(d)) if !d.cfg.is_empty() => {
+                        let mut new_attr = DocAttribute::default();
+                        new_attr.cfg = d.cfg.clone();
+                        Some(Attribute::Parsed(AttributeKind::Doc(Box::new(new_attr))))
```

Comment:
Why are we doing this weird clone of only the cfg condition here? Performance?
(If so, please add a comment explaining it)